daemon/cmd/daemon_main.go

```diff
@@ -456,6 +456,9 @@ func init() {
 	flags.String(option.K8sNamespaceName, "", "Name of the Kubernetes namespace in which Cilium is deployed in")
 	option.BindEnv(option.K8sNamespaceName)
 
+	flags.String(option.JoinK8sNamespaceName, "", "Name of the Kubernetes namespace join via kvstore registration")
+	option.BindEnv(option.JoinK8sNamespaceName)
+
```

Comment:
Why not reuse the `option.K8sNamespaceName` flag? That flag is never going to be used in VMs.

Reply:
Was doing that, but then needed an enable/disable flag for the agent registration, so decided to go with this. Agent needs to know it is a VM :-)

Reply:
Changed to add `--join-cluster` options, to be reused with `--k8s-namespace` option (unless the namespace label should not be set).